node/authorship/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Modified implementation of the basic block-authorship logic from
+// https://github.com/paritytech/substrate/tree/master/client/basic-authorship.
+// The block proposer explicitly pushes the `pallet_gear::run`
+// extrinsic at the end of each block.
+
+mod authorship;
+
+// #[cfg(test)]
+// mod tests;
```

Comment:
```suggestion
#[cfg(test)]
mod tests;
```

Also waiting for test you mentioned in dm